File: src/enemy.rs
pub struct Enemy {
    pub health: i64,
    pub max_health: i64,
    pub name: String,
    pub xp: u64,
    pub gold: u64,
}

impl Enemy {
    pub fn new() -> Self {
        Enemy {
            health: 50,
            max_health: 50,
            xp: 10,
            name: String::from("Skeleton"),
            gold: 1,
        }
    }

    pub fn health_percentage(&self) -> u16 {
        (self.health as f32 / self.max_health as f32 * 100.0) as u16
    }
}

File: idleslayer/Data/Enemy.cs
namespace idleslayer;

public class Enemy
{

    public string Name { get; set; } = "Skeleton";
    public float Health { get; set; } = 10;
    public float HealthMax { get; set; } = 10;
    public int Gold { get; set; } = 1;


    public Enemy()
    {
        Health = 10;
        HealthMax = 10;
        Gold = 1;
    }
    public Enemy(string name, float health, int gold)
    {
        Name = name;
        HealthMax = (float)Math.Ceiling(DataProcessor.ExponentialGrowth(health, 1, 0.15f));
        Health = HealthMax;
        Gold = gold;
    }

    public void Reset()
    {
        Health = HealthMax;
    }

    


}
